training/datastore/get_keywords.py

```diff
@@ -118,22 +118,22 @@ def from_github(self, package, url_df, api_url, api_token):
         else:
             return keywords
 
-    def find_keywords(self, df_, list_):
+    def find_keywords(self, data_, list_):
         """Find the keywords for given list of list of raw data."""
         package_lst = self.utility.flatten_list(list_)
         out_lst = list()
         for i in package_lst:
-            pkg_kwd_lst = list()
-            pkg_kwd_lst = self.utility.make_list_from_series(
-                self.from_existing_df(df_, i))
-            if not pkg_kwd_lst or type(pkg_kwd_lst[2]) != list:
-                logger.info("Finding from the NPM repository.")
-                pkg_kwd_dict = self.from_npm_registry(i)
-                pkg_kwd_lst = list(pkg_kwd_dict.values())
-                if len(pkg_kwd_lst[2]) == 0:
-                    logger.info("Trying to fetch from Github")
-                    api_url = 'https://api.github.com/graphql'
-                    api_token = self.get_data.github_token
-                    pkg_kwd_lst[2] = self.from_github(i, df_, api_url, api_token)
+            pkg_kwd_lst = [i, '', [], []]
```

Comment:
Can this be a NamedTuple?

Reply:
This is local list which is getting converted to `DataFrames` in next line (just after the loop).
`pd.DataFrame(out_lst, columns=['name', 'description', 'keywords', 'dependencies'])`
However, I will convert it into named tuple for better code readability.